mcs/class/System.Private.CoreLib/System.Runtime.CompilerServices/RuntimeHelpers.cs

```diff
@@ -1,7 +1,113 @@
 namespace System.Runtime.CompilerServices
 {
-	partial class RuntimeHelpers
+	public static class RuntimeHelpers
 	{
+		public delegate void TryCode (Object userData);
+
+		public delegate void CleanupCode (Object userData, bool exceptionThrown);
+
+		public static void InitializeArray (Array array, RuntimeFieldHandle fldHandle)
+		{
+			if (array == null || fldHandle.Value == IntPtr.Zero)
+				throw new ArgumentNullException ();
+
+			InitializeArray (array, fldHandle.Value);
+		}
+
+		public static extern int OffsetToStringData {
+			[MethodImpl (MethodImplOptions.InternalCall)]
+			get;
+		}
+
+		public static int GetHashCode (object o)
+		{
+			return Object.InternalGetHashCode (o);
+		}
+
+		public static new bool Equals (object o1, object o2)
+		{
+			if (o1 == o2)
+				return true;
+
+			if (o1 == null || o2 == null)
+				return false;
+
+			if (o1 is ValueType)
+				return ValueType.DefaultEquals (o1, o2);
+
+			return false;
+		}
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		public static extern object GetObjectValue (object obj);
+
+		public static void RunClassConstructor (RuntimeTypeHandle type)
+		{
+			if (type.Value == IntPtr.Zero)
+				throw new ArgumentException ("Handle is not initialized.", "type");
+
+			RunClassConstructor (type.Value);
+		}
+
+		public static void EnsureSufficientExecutionStack ()
+		{
+			if (SufficientExecutionStack ())
+				return;
+
+			throw new InsufficientExecutionStackException ();
+		}
+
+		public static bool TryEnsureSufficientExecutionStack ()
+		{
+			return SufficientExecutionStack ();
+		}
+
+		public static void ExecuteCodeWithGuaranteedCleanup (TryCode code, CleanupCode backoutCode, Object userData)
+		{
+		}
+
+		public static void PrepareConstrainedRegions ()
+		{
```

Comment:
It would be nice to have a comment that says that these methods are intentionally empty. Alternatively they could probably be moved to the shared CoreLib partition since the (empty) implementation will likely be identical across runtimes (https://github.com/dotnet/coreclr/issues/17903).

Reply:
I have the shared PR unification somewhere but the coverage is quite low (so low priority for now)